### PR TITLE
test/datatype: make typename check consistently

### PR DESCRIPTION
**Pull request by minsii**
The datatype/typename.c test compared name with output namelen for some
datatypes. Error cannot be detected if namelen is zero. This patch uses
MPI_MAX_OBJECT_NAME in the name string comparison similar to the
remaining typename check calls. It also prints output namelen when error
occurs.

### CODE REVIEW

**Comment by minsii** [comment]
test:jenkins/ch4/most

**Comment by raffenet** [comment]
This looks OK to me. Re-testing after rebase.
test:jenkins/ch4/most

**Comment by minsii** [comment]
I am not sure whether it is correct to completely replace `MPI_LONG_LONG_INT` with `MPI_LONG_LONG` in MPICH as they are synonym, spec says "Named predefined datatypes have the default names of the datatype name." 

We need to either change the test to allow the name to be either `MPI_LONG_LONG_INT` or `MPI_LONG_LONG`, or change MPICH to return `MPI_LONG_LONG_INT`.

**Comment by raffenet** [comment]
It seems MPICH used to support both names, but changed to only supporting one long ago: http://git.mpich.org/mpich-CVS.git/commitdiff/8bc3508298ba3bc6e9bef2467406ab01738c0c0a

I think the current interpretation is backwards, though. At the very least, we should return `MPI_LONG_LONG_INT` rather than always returning the allowed alias.

**Comment by minsii** [comment]
Agreed. I will make the change in MPICH to return `MPI_LONG_LONG_INT` name.

**Comment by minsii** [comment]
test:jenkins/ch4/most

**Comment by minsii** [comment]
test:jenkins/ch3/most

**Comment by minsii** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

File: src/mpi/datatype/type_get_name.c [hunk 1]
```diff
@@ -55,10 +55,10 @@ static mpi_names_t mpi_names[] = {
     type_name_entry(MPI_FLOAT),
     type_name_entry(MPI_DOUBLE),
     type_name_entry(MPI_LONG_DOUBLE),
-    /* LONG_LONG_INT is allowed as an alias; we don't make it a separate
+    /* LONG_LONG is a synonym of LONG_LONG_INT; we don't make it a separate
      * type */
-/*    type_name_entry(MPI_LONG_LONG_INT), */
-    type_name_entry(MPI_LONG_LONG),
+/*    type_name_entry(MPI_LONG_LONG), */
+    type_name_entry(MPI_LONG_LONG_INT),
```
Comment (raffenet) [inline src/mpi/datatype/type_get_name.c]: After some more thought, I'm tempted to say drop this patch. I'm not sure there's cause enough to change 15 year old behavior. The test changes we made should be sufficient to resolve the issue we were seeing. What do you think?

Comment (minsii) [reply src/mpi/datatype/type_get_name.c]: We can only change the tests for now, but I am not sure if it is a correct MPI behavior that returns LONG_LONG for LONG_LONG_INT. Maybe we can check with forum to figure out ?

Comment (minsii) [reply src/mpi/datatype/type_get_name.c]: The tests needs to be further changed to cover `LONG_LONG_INT` with name `LONG_LONG`. I will delete the above patch.

Comment (raffenet) [reply src/mpi/datatype/type_get_name.c]: OK, sorry for the churn. We can certainly ask the forum what they think the correct behavior is in this case.

Comment (raffenet) [reply src/mpi/datatype/type_get_name.c]: I checked Open MPI and they return "MPI_LONG_LONG_INT" for both. Similarly, they return "MPI_C_COMPLEX" for MPI_C_FLOAT_COMPLEX. This matches up with my original interpretation of the standard. I think we should match what they do and return the canonical type name even for these synonym types.

Comment (minsii) [reply src/mpi/datatype/type_get_name.c]: good catch for `MPI_C_FLOAT_COMPLEX`. I will fix it in this PR. So I will not discard the above name change commit.

**Comment by minsii** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by minsii** [comment]
The `gather_big` failure in ch4/ofi should be irrelevant.

**Comment by raffenet** [comment]
Looks good to me.

**Pull request merged**